src/coprocessor/dag/expr/builtin_string.rs

```diff
@@ -953,6 +1046,182 @@ mod tests {
         assert_eq!(got, exp, "length(NULL)");
     }
 
+    #[test]
+    fn test_locate_2_args() {
+        let cases = vec![
+            ("bar", "foobarbar", 4i64),
+            ("xbar", "foobar", 0i64),
+            ("", "foobar", 1i64),
+            ("foobar", "", 0i64),
+            ("", "", 1i64),
+            ("好世", "你好世界", 2i64),
+            ("界面", "你好世界", 0i64),
+            ("b", "中a英b文", 4i64),
+            ("BaR", "foobArbar", 4i64),
+        ];
+
+        for (substr, s, exp) in cases {
+            let substr = Datum::Bytes(substr.as_bytes().to_vec());
+            let s = Datum::Bytes(s.as_bytes().to_vec());
+            let got = eval_func(ScalarFuncSig::Locate2Args, &[substr, s]).unwrap();
+            assert_eq!(got, Datum::I64(exp));
+        }
+
+        let null_cases = vec![
+            (Datum::Null, Datum::Bytes(b"foobar".to_vec()), Datum::Null),
+            (Datum::Bytes(b"bar".to_vec()), Datum::Null, Datum::Null),
+            (Datum::Null, Datum::Null, Datum::Null),
```

Comment:
Could you add cases like `SELECT LOCATE('',NULL);SELECT LOCATE(NULL,'');`?

Reply:
sure